test/e2e/tests/test_retain_policy.py

```diff
@@ -0,0 +1,105 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+# 	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+"""Integration tests for the Resource retain policy
+"""
+
+import pytest
+import logging
+import botocore
+
+from acktest.k8s import resource as k8s
+from acktest.resources import random_suffix_name
+from e2e import (
+    service_marker,
+    create_sagemaker_resource,
+    sagemaker_client,
+)
+
+
+from e2e.replacement_values import REPLACEMENT_VALUES
+
+RESOURCE_PLURAL = "notebookinstancelifecycleconfigs"
```

Comment:
Even better would be to put this in the `test_notebook_instance_lifecycle_config.py` file and then import it in this file.

Reply:
Currently trying that, but it's making the Notebook lifecycle test slightly more flakey, need to investigate.

Reply:
Solved the issue.